src/app/models/grid.model.ts

```diff
@@ -15,12 +16,29 @@ export class Grid {
                 this.cells[r][c] = new Cell(this, r, c);
             }
         }
+        this.outOfBoundsCell = new Cell(this, -1, -1);
     }
 
-    getCell(row: number, col: number): Cell{
+    getCell(row: number, col: number): Cell {
+        return this.getCellBounded(row, col);
+    }
+
+    getCellUnbounded(row: number, col: number): Cell {
         const normalizedRow = (row + this.rowCount) % this.rowCount;
         const normalizedCol = (col + this.columnCount) % this.columnCount;
 
         return this.cells[normalizedRow][normalizedCol];
     }
+
+    getCellBounded(row: number, col: number): Cell {
```

Comment:
We get only bounded cells, so this logic can be moved to `getCell()`, I don't see the use for the extra function